### ISSUE TITLE
WalletConnect Integration

### ISSUE THREAD

**Issue by tjvsx** [0xHabitat/habitat#72]
### ⚡️ Challenge:
Integrate WalletConnect (https://walletconnect.com/) with the Habitat app to improve accessibility for mobile and non-Metamask wallet users. 

### 🦋 Desired Outcome
When clicking the ‘connect’ button in the sidebar, a popup will appear where the user can select their wallet connection method. 
<img width="415" alt="MetaMask" src="https://user-images.githubusercontent.com/62122206/146693878-459f2d1d-ff59-4560-842e-ce6c9533f6df.png">
*** styling not relevant

On desktop, after clicking ‘WalletConnect’, a QR will appear that allows for external mobile wallet connection. 
<img width="502" alt="WalletConnect" src="https://user-images.githubusercontent.com/62122206/146693889-284ba421-d27d-47f6-b682-4736e519df66.png">
WalletConnect handles the following: Scanning the QR with the preferred mobile wallet will prompt a connection request on the mobile wallet, to which confirmation will grant access to the desktop UI. 

On mobile, after clicking ‘WalletConnect’, a menu of mobile wallets will appear.
![Choose your preferred wallet](https://user-images.githubusercontent.com/62122206/146693893-22c2eb2b-7684-4975-ab3e-86e96115d368.png)
WalletConnect handles the following: Selecting the preferred wallet opens the mobile wallet locally. The user confirms the connection request and returns to the browser page for full UI access. 

### 🎯 Action Points:
- Get a local instance of the habitat app front-end running. 
- Follow the WalletConnect docs to integrate desired UX. 

### 🐇 Additional Information 
Useful links:
- https://ethereum.stackexchange.com/questions/101587/can-i-implement-walletconnect-with-vanilla-javascript
- https://docs.walletconnect.com/quick-start/dapps/client

### 🏆 Reward + Contributor involved
1800 DAI

**Comment by biendeveloper** [comment]
Hi @tjvsx is it open to everyone?  I would like to work on this. 

also 
```
WalletConnect handles the following: Selecting the preferred wallet opens the mobile wallet locally. The user confirms the connection request and returns to the browser page for full UI access.
```
Apps won't show up for mobile web? I guess this is for some mobile app right? is there an example website where you see these apps?

**Comment by tjvsx** [comment]
Hey @biendeveloper , yes the bounty is currently open to everyone, let us know if you'd like to work on it and we can get you up to speed on the app. The screenshots were taken from app.uniswap.org. The wallet connection experience is meant to be as similar as possible to their's.

**Comment by biendeveloper** [comment]
Yes! I will start working on this :) let me know if something can help me speed up the development

also, have you considered using [web3modal](https://www.npmjs.com/package/web3modal) which is used by [scaffold-eth ](https://github.com/scaffold-eth/scaffold-eth) and other dapps 

![image](https://user-images.githubusercontent.com/91308782/147368504-0aaa60d0-1140-465d-8731-dc4337fde455.png)

**Comment by tjvsx** [comment]
Web3modal would be awesome! The more wallets compatible the better. One thing to be aware of is the app uses pure/vanilla js and ethers.js, so I'm not sure if it's a common integration or not

**Comment by tjvsx** [comment]
Hey @biendeveloper, I wanted to let you know our focus has shifted to other areas for the time being and we will be reassessing these bounties at a later date. we are experimenting with eip2535 diamond standard and rethinking some of our front-end as well. if you'd like to get involved, reach out in our discord server: https://discord.gg/N89hAFAN

**Comment by biendeveloper** [comment]
Sure i just sent a message there.  Let me know. I can pickup some tasks on the front end 😃
